lib/RecordIdsPlugin.js

```diff
@@ -4,14 +4,23 @@
 */
 "use strict";
 
+const path = require("path");
+
+const looksLikeAbsolutePath = (maybeAbsolutePath) => /^(?:[a-z]:\\|\/)/i.test(maybeAbsolutePath);
+
 function makeRelative(compiler, identifier) {
 	const context = compiler.context;
-	return identifier.split("|").map(str => str.split("!").map(str => path.relative(context, str)).join("!")).join("|");
+	return identifier.split("|")
```

Comment:
yeah, i do not particular like the makeRelative function, however using split with regexp loses the ability to join by whatever caused the split.
if that it not important I can do it, but that may lead to clashes no?